File: Week_02/590/n_ary_tree_postorder_traversal.go
package main

type Node struct {
	Val      int
	Children []*Node
}

func postorder(root *Node) []int {
	res := make([]int, 0)
	var postOrder func(node *Node)
	postOrder = func(node *Node) {
		if node != nil {
			for i := 0; i < len(node.Children); i++ {
				postOrder(node.Children[i])
			}
			res = append(res, node.Val)
		}
	}
	postOrder(root)
	return res
}

File: Week_02/590/n_ary_tree_postorder_traversal.py
from typing import List


class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        res = []

        def post_order(node: Node):
            if node is not None:
                for c in node.children:
                    post_order(c)
                res.append(node.val)

        post_order(root)
        return res
